tvApp/src/main/java/band/effective/office/tv/domain/model/notion/EmployeeInfoEntity.kt

```diff
@@ -14,47 +11,57 @@ class EmployeeInfoEntity(
     val photoUrl: String,
 )
 
-fun List<EmployeeInfoEntity>.processEmployeeInfo(): List<EmployeeInfoUI> {
-    val resultList = mutableListOf<EmployeeInfoUI>()
-    this.forEach { employee ->
-        if (employee.nextBirthdayDate.isNotBlank() && isCelebrationToday(employee.nextBirthdayDate)) {
-            resultList.add(
+fun List<EmployeeInfoEntity>.processEmployeeInfo(): List<EmployeeInfoUI> =
+    map { employee ->
+        listOfNotNull(
+            if (employee.nextBirthdayDate.isNotBlank() && isYearCelebrationToday(employee.nextBirthdayDate)) {
                 BirthdayUI(
                     employee.firstName,
                     employee.photoUrl,
                 )
-            )
-        }
-        if (employee.startDate.isNotBlank() && isCelebrationToday(employee.startDate)) {
-            resultList.add(
-                AnniversaryUI(
+            } else null,
```

Comment:
you can use 'MutableList' and not add null. Also, why create nested arrays and then create a linear one from them. This is solved by a 'mutableList' and 'when'

Reply:
Null will filter later

Reply:
We need to simplify this. Time for brainstorming

Reply:
I think create MutableList for mapping is not best practics

Reply:
why? The code will become more readable and there will be no dancing with tambourines with `flatten()` As a temporary solution, fits better than this

Reply:
> I think create MutableList for mapping is not best practics
why? The code will become more readable and there will be no dancing with tambourines with flatten() As a temporary solution, fits better than this

Reply:
I think it's readable. I define congratulations for all employee and then combine them into a common list. Muttable list not match functional style

Reply:
If I remember correctly, he used to use the `MutableList  `implementation before, but there was a problem with the order of the checks there. I suggest leaving it that way, but I'll keep a note for potential refactoring.